src/recurrer.ts

```diff
@@ -1,111 +1,155 @@
-import _ from 'lodash'
+import * as _ from 'lodash'
+import { ADDITIONAL_UNIT, DAYS_IN_WEEK, HOUR_IN_DAY, MIN_IN_HOUR, MS_IN_SEC, NO_DIFF, SEC_IN_MIN } from './constants'
 
-// For producing recurring timestamps 
-export default class Recurrer {
-  async getDailyRecurringTimestamps(
-    startTimestamp: number,
-    numberRecurring: number,
-    hourOfDay: HourOfDay,
-  ): Promise<number[]> {
+// For producing recurring timestamps
+export class Recurrer {
+  /**
+   * Input starting timestamp in milliseconds along with desired hour of day.
+   * Output up to 24 daily recurring timestamps in milliseconds
+   * @param startTimestamp
+   * @param numberRecurring
+   * @param hourOfDay
+   * @returns daily recurring timestamps
+   */
+  getDailyRecurringTimestamps(startTimestamp: number, numberRecurring: number, hourOfDay: HourOfDay): number[] {
     const startDate = new Date(startTimestamp)
     const startHour = startDate.getUTCHours()
     const startYear = startDate.getUTCFullYear()
     const startMonth = startDate.getUTCMonth()
-    const startDay = (startHour < hourOfDay) ? startDate.getUTCDate() : startDate.getUTCDate() + 1
+    const startDay = startHour < hourOfDay ? startDate.getUTCDate() : startDate.getUTCDate() + ADDITIONAL_UNIT
```

Comment:
I'm a little confused on what is happening in this line?  Does this line assume `startTimestamp` is the current date/time (ie. Date.now)?

Reply:
ahhh yes yes. sorry about this. i should probably fix the naming. this is basically trying to fix the following: 

say it's 5 PM on 5/2. you want to schedule recurring task for 1 PM daily. 1 PM has already passed for 5/2. so you want to schedule first day for 1 PM on 5/3. so ,we can only know that by checking the time that was input and see if that time was past the hourOfDay for the current day. if past hourOfDay, then we need to do +1 to the input date that the user input. otherwise, just use the same day

Reply:
but you are assuming the time that was input was today's date, right?  what if the time input is 1 PM on 6/2?

Reply:
oh true, good point. i should make it <= rather than <, just so if they give me exact timestamp, i don't bump the to the next day. but if it's <=, then if they input 1 PM at 6/2, then it should auto-accept. and similarly, if they choose a past timestamp, that's fine too, i'll generate teh times with this function all the same. the scheduler timestamp validation function will just reject it afterwards.

imo this function shouldn't care about what the _current_ day is. it just cares about the next available slot in accordance with the input timestamp. but that doest still require me to fix my < to <= to make this statement true i think haha 😅

Reply:
ok.  I think I understand better but worry about it being confusing for others.  Probably solvable with docs.

Pretty much, the `startTimestamp` says at this time we will start executing tasks on the `hourOfDay` provided.  If that hour has passed on the day provided in the `startTimestamp` then the first task will be executed on the following day.